bluesky_queueserver/manager/manager.py

```diff
@@ -609,35 +612,157 @@ async def zmq_server_comm(self):
+                    await asyncio.wait_for(self._fut_watchdog_comm,
+                                           timeout=self._timeout_watchdog_comm)
+                    response = self._fut_watchdog_comm.result()
+                    if "id" in response:
+                        if response["id"] != msg["id"]:
+                            logger.error("Response Watchdog->RE Manager contains incorrect ID: %s. Expected %s",
+                                         response["id"], msg["id"])
+                        del response["id"]
+                    else:
+                        logger.error("Response Watchdog->RE Manager contains no id: %s", pprint.pformat(response))
+                else:
+                    response = None
+                return response
+
+            finally:
+                self._event_watchdog_comm.clear()  # Clear before the exit as well
+
+    async def _watchdog_response(self, response):
+        """
+        Set the fututure with the results
+        """
+        if self._event_watchdog_comm.is_set():
+            self._event_watchdog_comm.clear()  # Clear once the message received
+            self._fut_watchdog_comm.set_result(response)
+        else:
+            logger.error("Unsolicited message received Watchdog->Re Manager: %s. Message is ignored",
+                         pprint.pformat(response))
+
+    # ===============================================================================
+    #         Functions that send commands/request data from Watchdog process
+
+    async def _watchdog_start_re_worker(self, timeout=0.5):
+        """
+        Initiate the startup of the RE Worker. Returned 'success==True' means that the process
+        was created successfully and RE environment initialization is started.
+        """
+        try:
+            response = await self._watchdog_send("start_re_worker")
+            success = response["result"]["success"]
+        except asyncio.TimeoutError:
+            success = False
+        return success
+
+    async def _watchdog_join_re_worker(self, timeout=0.5):
+        """
+        Request Watchdog to join RE Worker process. The sequence of orderly closing of the process
+        needs to be initiated before attempting to join the process.
+        """
+        try:
+            response = await self._watchdog_send("join_re_worker", timeout=timeout)
+            success = response["result"]["success"]
+        except asyncio.TimeoutError:
+            success = False
+        return success
+
+    async def _watchdog_kill_re_worker(self):
+        """
+        Request Watchdog to kill RE Worker process (justified only if the process is not responsive).
+        """
+        try:
+            response = await self._watchdog_send("kill_re_worker")
+            success = response["result"]["success"]
+        except asyncio.TimeoutError:
+            success = False
+        return success
```

Comment:
The 3 functions above are almost identical except the arguments passed to `self._watchdog_send(...)`. Maybe it's worth to generalize?

Reply:
I think we should keep it for now. This is temporary solution. Those functions may diverge in the next iteration, because timeout handling could be (or should be) different for those cases. If not, then I will generalize it later.

Reply:
Sounds good.